code/modules/mob/living/simple_animal/hostile/mushroom.dm

```diff
@@ -1,205 +0,0 @@
-			if(level_gain < 1)//So we still gain a level if two mushrooms were the same level
-				level_gain = 1
-			shroom.LevelUp(level_gain)
-		shroom.adjustBruteLoss(-shroom.maxHealth)
-		qdel(src)
-		return TRUE
-	return ..()
-
-/mob/living/simple_animal/hostile/mushroom/revive(full_heal_flags = NONE, excess_healing = 0, force_grab_ghost = FALSE)
-	. = ..()
-	if(!.)
-		return
-
-	icon_state = "mushroom_color"
-	UpdateMushroomCap()
-
-/mob/living/simple_animal/hostile/mushroom/death(gibbed)
-	. = ..()
-	UpdateMushroomCap()
-
-/mob/living/simple_animal/hostile/mushroom/proc/UpdateMushroomCap()
-	cut_overlays()
-	cap_living.color = cap_color
-	cap_dead.color = cap_color
-	if(health == 0)
-		add_overlay(cap_dead)
-	else
-		add_overlay(cap_living)
-
-/mob/living/simple_animal/hostile/mushroom/proc/Recover()
-	visible_message(span_notice("[src] slowly begins to recover."))
-	faint_ticker = 0
-	revive(HEAL_ALL)
-	UpdateMushroomCap()
-	recovery_cooldown = 1
-	addtimer(CALLBACK(src, PROC_REF(recovery_recharge)), 300)
-
-/mob/living/simple_animal/hostile/mushroom/proc/recovery_recharge()
-	recovery_cooldown = 0
-
-/mob/living/simple_animal/hostile/mushroom/proc/LevelUp(level_gain)
-	if(powerlevel <= 9)
-		powerlevel += level_gain
-		if(prob(25))
-			melee_damage_lower += (level_gain * rand(1,5))
-		else
-			melee_damage_upper += (level_gain * rand(1,5))
-		maxHealth += (level_gain * rand(1,5))
-	adjustBruteLoss(-maxHealth) //They'll always heal, even if they don't gain a level, in case you want to keep this shroom around instead of harvesting it
-
-/mob/living/simple_animal/hostile/mushroom/proc/Bruise()
-	if(!bruised && !stat)
-		src.visible_message(span_notice("The [src.name] is bruised!"))
-		bruised = 1
-
-/mob/living/simple_animal/hostile/mushroom/attackby(obj/item/I, mob/user, params)
-	if(istype(I, /obj/item/food/grown/mushroom))
-		if(stat == DEAD && !recovery_cooldown)
-			Recover()
-			qdel(I)
-		else
-			to_chat(user, span_warning("[src] won't eat it!"))
-		return
-	if(I.force)
-		Bruise()
-	..()
-
-/mob/living/simple_animal/hostile/mushroom/attack_hand(mob/living/carbon/human/user, list/modifiers)
-	..()
-	if(user.combat_mode)
-		Bruise()
-
-/mob/living/simple_animal/hostile/mushroom/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum)
-	..()
-	if(isitem(AM))
-		var/obj/item/T = AM
-		if(T.throwforce)
-			Bruise()
-
-/mob/living/simple_animal/hostile/mushroom/bullet_act(obj/projectile/P)
-	. = ..()
-	if(P.damage > 0 && P.damage_type == BRUTE)
-		Bruise()
-
-/mob/living/simple_animal/hostile/mushroom/harvest()
-	var/counter
-	for(counter=0, counter <= powerlevel, counter++)
-		var/obj/item/food/hugemushroomslice/S = new /obj/item/food/hugemushroomslice(src.loc)
-		S.reagents.add_reagent(/datum/reagent/drug/mushroomhallucinogen, powerlevel)
-		S.reagents.add_reagent(/datum/reagent/medicine/omnizine, powerlevel)
-		S.reagents.add_reagent(/datum/reagent/medicine/synaptizine, powerlevel)
```

Comment:
Why did you omit this whole part? `harvest()` is a proc on `/mob/living`? You should move it over and get rid of the single-letter variable `S` on line 198 here.